### ISSUE TITLE
render again

### ISSUE THREAD

**Issue by crper** [reach/router#98]
- react@16.4.1
- styled-components
- @reach/router @1.1.1

**Code**

```javascript
import React, { PureComponent } from 'react';
import { Router, Location } from '@reach/router';
import styled from 'styled-components';
import { TransitionGroup, CSSTransition } from 'react-transition-group';
const FadeCss = styled.div`
    .pagefade-enter,
    .pagefade-appear {
        opacity: 0;
    }
    .pagefade-enter.pagefade-enter-active,
    .pagefade-appear.pagefade-appear-active {
        opacity: 1;
        transition: opacity 1000ms;
    }
    .pagefade-enter.pagefade-enter-active {
        transition-delay: 1000ms;
    }
    .pagefade-exit {
        opacity: 1;
    }
    .pagefade-exit.pagefade-exit-active {
        opacity: 0;
        transition: opacity 1000ms;
    }

    animation-duration: inherit;
`;

const FadeTransition = props => {
    console.log(props);
    return (
        <FadeCss>
            <Location>
                {({ location }) => (
                    <TransitionGroup>
                        <CSSTransition
                            key={location.key}
                            classNames="fade"
                            timeout={{ enter: 300, exit: 300 }}
                            className="pagefade"
                            mountOnEnter={true}
                            unmountOnExit={true}>
                            <Router location={location} className="router">
                                {props.children}
                            </Router>
                        </CSSTransition>
                    </TransitionGroup>
                )}
            </Location>
        </FadeCss>
    );
};

export default FadeTransition;


// ---------- Parent Component--------
 <FadeTransition>
                <LoginPage path="login" default />
                <RegisterPage path="register" />
                <ForgetPasswordPage path="forgetpassword" />
</FadeTransition>


```

such as gif ,

![2018-07-12 13 46 05](https://user-images.githubusercontent.com/8272877/42614792-fba7a9a0-85d9-11e8-93e6-ae792761d15e.gif)

if`   timeout={{ enter: 300, exit: 300 }}`

![2018-07-12 13 39 04](https://user-images.githubusercontent.com/8272877/42614810-0f151252-85da-11e8-8ed6-24ff60a673c6.gif)

I need help , Thanks

**Comment by crper** [comment]
closed

**Comment by crper** [comment]
closed #98
